service/common/src/main/java/org/apache/polaris/service/catalog/PolarisCatalogHandlerWrapper.java

```diff
@@ -1019,88 +1016,86 @@ public void commitTransaction(CommitTransactionRequest commitTransactionRequest)
       throw new BadRequestException("Cannot update table on external catalogs.");
     }
 
-    if (!(baseCatalog instanceof BasePolarisCatalog)) {
+    if (baseCatalog instanceof BasePolarisCatalog basePolarisCatalog) {
```

Comment:
Is it totally forbidden to do `if (!(baseCatalog instanceof BasePolarisCatalog))`? it looks like a overkill to me :thinking:

Reply:
With the new errorprone version, yes.
Looks like the behavior to rely on errorprone defaults was there since forever :shrug:

Reply:
Blocking this particular coding option does not sound good to me. I believe it is quite valid to use negative `instanceof` checks in `if` conditions.

Reply:
As an alternative we could use [the config-file approach](https://github.com/projectnessie/nessie/blob/main/codestyle/errorprone-rules.properties).

Reply:
Hm, while looking into the config-file approach, I realized that exactly this pattern is explicitly set to `ERROR` [here](https://github.com/apache/polaris/blob/1cf9e082e851deaf7e346af0801fa436308734fb/build-logic/src/main/kotlin/polaris-java.gradle.kts#L51) via #393

Reply:
Created #1233.

I don't mind nuking this PR, setting it to draft for now, unless there's interest in this change.